### ISSUE TITLE
Bug: Connected Tileset can't be deleted.

### ISSUE THREAD

**Issue by UliAbo** [mapeditor/tiled#3413]
**Describe the bug**
A Tileset can't be deleted anymore, the button has no function / is greyed out:
![image](https://user-images.githubusercontent.com/1249266/179248955-ae3ef822-9390-4299-bdfb-9d74247b498e.png)

**To Reproduce**
I've changed, deleted and renamed lots of Tilesets (and properly didn't delete and rename them correctly). So I can't reproduce it, but I can send you the file if necessary.

**Specifications:**
 - OS: Win10 64bit
 - Tiled Version: v1.9

**Comment by eishiya** [comment]
That button does not *delete* the tileset, it *removes* it from the current map. Are you sure this tileset is in the map in the first place? It may just be displayed because it's present in a different map, or is open as a document in Tiled.

Does the tileset still appear if you restart Tiled?

**Comment by UliAbo** [comment]
Yes, restarting Tiled doesn't help. 
It's not opened as a document and nowhere used in any map of the project. 

When it would have been used I would have gotten prompted if I would really like to remove all references, etc. But I don't get asked this.

And I can now reproduce this bug anytime I would like to remove a tileset :(

**Comment by eishiya** [comment]
Since you can reproduce this now, could you perhaps make a minimal example to share here?

**Comment by UliAbo** [comment]
Yes, I keep on testing, I already found further bugs, so maybe I can make then a one file upload where several bugs can be experienced.

**Comment by eishiya** [comment]
One bug at a time is usually easier to test :] An example with several bugs is not a minimal example.

**Comment by bjorn** [comment]
@UliAbo Did you try closing all files, and then just opening a single map? Is the tileset there, with this button still disabled? As long as you have multiple maps open, it'll be harder to be sure that indeed none of these maps fact refer to this tileset.

**Comment by UliAbo** [comment]
I tried to reproduce it, I can't unfortunately! Then let's close this for now.
